quantinuum-hugr/src/builder/circuit.rs

```diff
@@ -208,4 +294,54 @@ mod test {
 
         assert_matches!(build_res, Ok(_));
     }
+
+    #[test]
+    fn ancillae() {
+        let build_res = build_main(
+            FunctionType::new(type_row![QB], type_row![QB]).into(),
+            |mut f_build| {
+                let mut circ = f_build.as_circuit(f_build.input_wires());
+                assert_eq!(circ.n_wires(), 1);
+
+                let [q0] = circ.tracked_units_arr();
+                let [ancilla] = circ
+                    .append_with_outputs::<CircuitUnit>(q_alloc(), [])?
+                    .try_into()
+                    .expect("Expected a single ancilla wire");
```

Comment:
Maybe we should have a panicking `as_arr()` utility on `Outputs` too? Can be follow up

Reply:
https://github.com/CQCL/hugr/pull/871